dts/arm/silabs/efr32mg24b310f1536im48.dtsi

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright (c) 2019 Steven Lemaire
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <mem.h>
+#include <silabs/efr32mg24.dtsi>
+
+/ {
+	sram0: memory@20000000 {
+		reg = <0x20000000 DT_SIZE_K(256)>;
+	};
+
+	soc {
+		compatible = "silabs,efr32mg24b310f1536im48",
+					 "silabs,efr32mg24", "silabs,efr32",
+					 "simple-bus";
+
+		flash-controller@50030000 {
+			flash0: flash@0 {
+				reg = <0 DT_SIZE_K(1536)>;
```

Comment:
Without having the [FLASH_BASE_ADDRESS](https://docs.zephyrproject.org/latest/kconfig.html#CONFIG_FLASH_BASE_ADDRESS) set in [zephyr,flash](https://docs.zephyrproject.org/latest/build/dts/api/api.html), `west flash` (using the jlink runner) refuses to program the device for me due to the HEX file starting at address 0.

Shouldn't these lines consider the flash memory offset of 0x08000000 (cf. [figure 3.2. EFR32MG24 Memory Map](https://www.silabs.com/documents/public/data-sheets/efr32mg24-datasheet.pdf#page=20))? Changing the device tree like this allows me to successfully use `west flash`:

```suggestion
		flash-controller@50030000 {
			flash0: flash@8000000 {
				reg = <0x08000000 DT_SIZE_K(1536)>;
```

Reply:
Hi, I can't seem to reproduce what you're describing here, i.e. west flashes the board normally without any modifications. Could you describe your env in more detail, e.g. OS, JLink ver, west ver etc., and paste the error message?